ldes-server-infra-mongo/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/infra/mongo/LdesFragmentMongoRepository.java

```diff
@@ -31,6 +32,6 @@ public Optional<LdesFragment> retrieveLastFragment(String view) {
         return repository.findAll().stream()
                 .filter(ldesFragmentEntity -> !ldesFragmentEntity.isImmutable())
                 .map(LdesFragmentEntity::toLdesFragment)
-                .findFirst();
+                .min(Comparator.comparing(LdesFragment::getFragmentId));
```

Comment:
Seems like my comments would also fit the previous version too, but isn't this the opposite of what our method says?

I'd expect that we sort on the fragment value in descending order?

Reply:
Indeed, we should go with something like retrieveFirstOpenFragment()
I'll add a test :-) That should clarify the behaviour.

Reply:
Well i think we want to retrieve the last fragment no? Need to add more test cases indeed

Reply:
The more I think about it first and last don't really mean anything. In time based fragmentation those words have a value, but in a general LDES one would expect that based on the fragmentation there's only one open fragment. Called the method retrieveOpenFragment() for now. And added tests ;-)